crates/solvers/src/infra/cli.rs

```diff
@@ -17,6 +18,10 @@ pub struct Args {
     #[arg(long, env, default_value = "127.0.0.1:7872")]
     pub addr: SocketAddr,
 
+    /// The chain ID this solver is for.
+    #[arg(long, env, value_parser = parse_chain_id)]
+    pub chain_id: eth::ChainId,
```

Comment:
I think this should also be specified in the config file. Why would you specify the WETH address in config but specify this in the CLI? Where do you draw the line?

I'd say everything except the address and log level should go to the config file.

Reply:
> Why would you specify the WETH address in config but specify this in the CLI?

To me the chain ID is similar to the Ethereum RPC configuration, in that it is for the `solvers` binary and not for a specific solver subcommand.

That being said, you can imagine a "pure" solver that doesn't care about this and only uses passed in liquidity, so I think you are right that it should go in the configuration file.

Reply:
0016437aa21683b664984c912239a576cd9e945a